File: Python/InsertionSort.py
def InsertionSort(arr):
    for i in range(1,len(arr)):
        for j in range(i,0,-1):
            if(arr[j]<arr[j-1]):
                temp=arr[j]
                arr[j]=arr[j-1]
                arr[j-1]=temp
    return



mylist=[]
j=100000
for i in range(10000):
    mylist.append(j)
    j-=1


InsertionSort(mylist)
print("\nSorted list : ")
print(mylist)

File: Javascript/InsertionSort.js
var arr = [34,21,2,6,7,89,44,0,-34];



function InsertionSort(arr)
{
    for(var i=1;i<arr.length;i++)
    {
        var value = arr[i];
        var j = i-1;
        while((j>=0)&&(arr[j]>value))
        {
            arr[j+1]=arr[j];
            j--;
        }

        arr[j+1]=value;
    }
    return arr;
}

var myarr = InsertionSort(arr);
console.log(myarr);